src/textual/actions.py

```diff
@@ -32,6 +42,21 @@ def parse(action: str) -> tuple[str, tuple[Any, ...]]:
     )
 
 
+def _process_dynamic_event_params(action_params_str: str, event: Event) -> str:
```

Comment:
It works but it is definitely a bit hacky to be honest - and also limited, as the only things we can inject with this mini language are scalar values.

On one hand it certainly limits the potential security issues of this feature (if the keybinding is taken from a database for example, and contains sequences such as `$event.__class__.os.environ` to expose the environment variables the app is running with )

But on the other hand it prevents passing the Event instance itself for example, which I guess could be handy if we wanted to do checks such as this in the event handler: 
```py
def action_on_digit(self, key:str, sender):
  if sender is left_panel_widget: # <-- not possible, since we cannot pass objects with this syntax
     do_something() 
```
We can always inject is the string representation of the sender's `__class__` with this mini-language, but it's more limited.

Ah, the toughness of trade-offs... :sweat_smile:  ¯\_(ツ)_/¯